src/httpAuthScheme.ts

```diff
@@ -370,6 +370,75 @@ export class Token {
         return token;
     }
 
+    static parse(tokenTypeEntry: TokenTypeEntry, data: string): Token {
+        // Consumes data:
+        //   challenge="abc...", token-key="123..."
+
+        const attributes = data.split(',');
+        let ppToken = undefined;
+
+        for (const attr of attributes) {
+            const idx = attr.indexOf('=');
+            let attrKey = attr.substring(0, idx);
+            let attrValue = attr.substring(idx + 1);
+            attrValue = attrValue.replaceAll('"', '');
+            attrKey = attrKey.trim();
+            attrValue = attrValue.trim();
+
+            switch (attrKey) {
+                case 'token': {
+                    const tokenEnc = base64url.parse(attrValue);
+                    ppToken = Token.deserialize(tokenTypeEntry, tokenEnc);
+                    break;
+                }
+            }
+        }
+
+        // Check for mandotory fields.
+        if (ppToken === undefined) {
+            throw new Error('cannot parse token');
+        }
+
+        return ppToken;
+    }
+
+    static parseMultiple(tokenTypeEntry: TokenTypeEntry, header: string): Token[] {
+        // Consumes data:
+        //   PrivateToken token="abc...",
+        //   PrivateToken token=def...
+        const challenges = parseWWWAuthenticate(header);
+
+        const listTokens = new Array<Token>();
+
+        for (const challenge of challenges) {
+            if (!challenge.startsWith('PrivateToken ')) {
+                continue;
+            }
+            const chl = challenge.slice('PrivateToken '.length);
+            const privToken = Token.parse(tokenTypeEntry, chl);
+            listTokens.push(privToken);
+        }
+
+        return listTokens;
+    }
+
+    toString(quotedString = false): string {
+        // WWW-Authenticate does not impose authentication parameters escape with a double quote
+        // For more details, refer to RFC9110 Section 11.2 https://www.rfc-editor.org/rfc/rfc9110#section-11.2
+        const quote = quotedString ? '"' : '';
+        // eslint-disable-next-line func-style
+        const authParamToString = (param: string, value: string | number): string =>
+            `${param}=${quote}${value}${quote}`;
+        const token = base64url.stringify(this.serialize());
+        const authParams: Record<string, string | number> = {
+            token,
+        };
+        const params = Object.entries(authParams)
+            .map(([param, value]) => authParamToString(param, value))
+            .join(', ');
+        return `PrivateToken ${params}`;
```

Comment:
it is simpler to do:

```suggestion
        const token = base64url.stringify(this.serialize());
        return `PrivateToken token=${quote}${token}${quote}`;
```

Reply:
I moved the `toString` function to rfc9110.ts, and updated both Token and PrivateToken methods to use it.